### PR TITLE
Zoltan2: Sphynx updates

### PR DESCRIPTION
**Pull request by jennloe**
@trilinos/zoltan2 

## Motivation
Adding drivers and utilities previously developed by @seheracer in her research using Sphynx. 
Recently, it has become clear that these files need to be added to the main Trilinos codebase for accessibility and maintainability. 

`largestComponent2Binary.cpp` Provides a utility that extracts the largest component from a graph matrix (in a matrix market file) and converts the file to binary form.
`readMatrixFromBinaryFile.cpp` reads the corresponding binary files. 
`Sphynx_Research_Driver.cpp` provides extensive options to be used in testing Sphynx.

Additionally, we have modified `Zoltan2_Sphynx.hpp` and `Zoltan2_SphynxProblem.hpp` to allow for use of an upcoming randomized eigensolver that will soon be available in Anasazi.  Additionally, we have added a function so that users can ask Sphynx to partition a graph based upon user-provided eigenvectors.

Finally, we have improved the formatting of several files.
These additions will support completion of upcoming ECP milestones.

### CODE REVIEW

**Review by cgcgcg — DISMISSED** [review]
Please make sure all commits have a valid author.

File: packages/zoltan2/sphynx/src/Zoltan2_Sphynx.hpp [hunk 1]
```diff
@@ -422,30 +427,37 @@ namespace Zoltan2 {
...
+          newVal(rowOffsets(i) + diagOffsets(i)) = rowOffsets(i+1) - rowOffsets(i) - 1;
+          }
+          );
       Kokkos::fence ();
 
       // Create the Laplacian maatrix using the input graph and with the new values
       Teuchos::RCP<matrix_t> laplacian (new matrix_t(graph_, newVal));
       laplacian->fillComplete (graph_->getDomainMap(), graph_->getRangeMap());
 
+      // Create the Laplacian maatrix using the input graph and with the new values
```
Comment (cgcgcg) [inline packages/zoltan2/sphynx/src/Zoltan2_Sphynx.hpp:440]: spelling

**Comment by jennloe** [comment]
> Please make sure all commits have a valid author.

I'm planning to squash merge the commits.  There's too many, and most aren't independently meaningful.  Does this still make a difference in that case? (Actually, I'm not even sure how I ended up with commits that have no author! Don't think I've managed that before!!)

File: packages/zoltan2/sphynx/src/Zoltan2_Sphynx.hpp (hunk 1 above)
Comment (jennloe) [reply packages/zoltan2/sphynx/src/Zoltan2_Sphynx.hpp]: Fixed

**Review by egboman — APPROVED** [review]
Looks good! Most of these new features are for developers only, but good to preserve these files in the repo. Thanks, @jennloe !

**Pull request merged**